Source/Caches/HACachedStates.swift

```diff
@@ -6,26 +6,24 @@ public extension HACachesContainer {
 /// Cached version of all entity states
 public struct HACachedStates {
     /// All entities
-    public var all: Set<HAEntity>
+    public var all: Set<HAEntity> = []
     /// All entities, keyed by their entityId
     public subscript(entityID: String) -> HAEntity? {
         get { allByEntityId[entityID] }
         set { allByEntityId[entityID] = newValue }
     }
 
     /// Backing dictionary, whose mutation updates the set
-    private var allByEntityId: [String: HAEntity] {
+    public var allByEntityId: [String: HAEntity] {
         didSet {
             all = Set(allByEntityId.values)
         }
     }
 
     /// Create a cached state
-    /// - Parameter entities: The entities to start with
-    public init(entities: [HAEntity]) {
-        self.all = Set(entities)
-        self.allByEntityId = entities.reduce(into: [:]) { dictionary, entity in
-            dictionary[entity.entityId] = entity
-        }
```

Comment:
minor: I think this might be worth keeping around if only so the tests are cleaner, too.